### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by Otegaa**
###In this PR,
- I installed redux-toolkit
- Configured store.js
- Defined slice for books
- Added a reducer for adding book and one for removing book in book slice
- Defined slice for categories
- Added a reducer for categories that returns "Under construction" string

### CODE REVIEW

File: src/redux/app/store.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { configureStore } from '@reduxjs/toolkit';
```
Comment (danmainah) [inline src/redux/app/store.js]: - [ ] @Otegaa Great job adding all the files required, however, you didn't follow the ducks' pattern when creating them, you have extra folders features and app which shouldn't be there, your structure should match the photo below 

image

File: src/redux/features/books/BookSlice.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    addBook: (state, { payload }) => {
+      state.books.push(payload);
+    },
+    removeBooks: (state, { payload }) => {
+      state.books = state.books.filter((book) => book.id !== payload);
+    },
+  },
+});
+
+export const { addBook } = bookSlice.actions;
```
Comment (danmainah) [inline src/redux/features/books/BookSlice.js]: - [ ] @Otegaa great work on your book reducer however you forgot to export removeBooks action

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Otegaa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Github flow was correctly used
✔️ Good Professional README file.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by pinkmoon25 — APPROVED** [review]
Hi @Otegaa , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
Linter checks :heavy_check_mark: 
Gitflow :heavy_check_mark: 
Changes Implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**